### PR TITLE
Python 3.x & Windows compatibility

### LINKED ISSUE #6: python3
Fancy a patch for python3 and systems that python doesn't default to "utf8" encoding?

Otherwise the following errors are generated

### Python2/3 print

```
$ python deletetweets.py -d 2013-10-01
  File "deletetweets.py", line 30
    print "Deleting tweet #{0} ({1})".format(tweet_id, tweet_date)
                                    ^
SyntaxError: invalid syntax
```

### UTF-8 windows bug

```
$ python deletetweets.py -d 2015-12-31
Traceback (most recent call last):
  File "deletetweets.py", line 63, in <module>
    main()
  File "deletetweets.py", line 59, in main
    delete(api, args.date, args.restrict)
  File "deletetweets.py", line 18, in delete
    for row in csv.DictReader(file):
  File "C:\Program Files\Python36\lib\csv.py", line 111, in __next__
    self.fieldnames
  File "C:\Program Files\Python36\lib\csv.py", line 98, in fieldnames
    self._fieldnames = next(self.reader)
  File "C:\Program Files\Python36\lib\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 2268: character maps to <undefined>
```

### PR DESCRIPTION
**Pull request by Lewiscowles1986**
Fixes #6

~should be backwards compatible with Python 2~ & windows & non-windows systems

### CODE REVIEW

File: deletetweets.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 __version__ = "0.1"
 
 def delete(api, date, r):
-    with open("tweets.csv") as file:
+    with open("tweets.csv", encoding="utf8") as file:
```
Comment (koenrh) [inline deletetweets.py]: Shouldn't the value be "utf-8" (including the dash)?

Comment (Lewiscowles1986) [reply deletetweets.py]: it's not bc with python2 universally, unfortunately it gets a bit funny. Also "utf8" and "utf-8" should be synonymous. I've used this to delete over 17,000 tweets

**Comment by koenrh** [comment]
Could you import the 'io' module by adding `import io` at the top, and then replace `open` with `io.open`? That seems to work with both Python 2.7 and 3.x.

**Comment by Lewiscowles1986** [comment]
I have a test branch with that. I can of course include it, but it had some issues with the utf8 encoding for me because of how the csv file is loaded

**Comment by koenrh** [comment]
@Lewiscowles1986 Could you include it in this branch? I have tested it locally, and it seems to work just fine with both Python 2.7 and 3.x. Or could you otherwise provide an example CSV file that is causing issues? I'm happy to look into it.

**Comment by Lewiscowles1986** [comment]
Here's the message

```
python2 deletetweets.py -d 2019-01-01
Deleting tweet #1021319911627468801 (2018-07-23)
Exception: [{u'message': u'No status found with that ID.', u'code': 144}]

Deleting tweet #1021309798481047552 (2018-07-23)
Exception: [{u'message': u'No status found with that ID.', u'code': 144}]

Deleting tweet #1021300629388578816 (2018-07-23)
Exception: [{u'message': u'No status found with that ID.', u'code': 144}]

Traceback (most recent call last):
  File "deletetweets.py", line 63, in <module>
    main()
  File "deletetweets.py", line 60, in main
    delete(api, args.date, args.restrict)
  File "deletetweets.py", line 19, in delete
    for row in csv.DictReader(file):
  File "C:\Python27\lib\csv.py", line 108, in next
    row = self.reader.next()
UnicodeEncodeError: 'ascii' codec can't encode character u'\u201c' in position 144: ordinal not in range(128)

```

the offending tweet (entry) seems to be 

```
"1021300629388578816","1021147106390798336","440625608","2018-07-23 07:46:59 +0000","<a href=""http://twitter.com"" rel=""nofollow"">Twitter Web Client</a>","@thinkfoo @bitfield Do you per-chance sell single-line dev-setups? They don't just exist, they need to be created and maintained. That might be your value-add skill that you bring aside from the thing they hired you for.","","","",""

```

Fairly innocuous https://www.fileformat.info/info/unicode/char/201c/index.htm

**Comment by koenrh** [comment]
Thanks!

**Comment by Lewiscowles1986** [comment]
No thank you. I'd been wondering what to do with my tweets. I think every time it gets to 1k I'll backup and delete using this tool, maybe move some of my favourites to a blog at some stage.

**Pull request merged**